### PR TITLE
Customer API Sample Application

### LINKED ISSUE #136: Add customer endpoint sample code
Enhance an existing sample app, or create a new one, to demonstrate how to create a customer, log in, reset password, and fetch order history.

### PR DESCRIPTION
**Pull request by dbart01**
#### What this lets you do
- create new customer account
- login with existing customer account
- view existing customer orders

Relies on #140 
Fixes #136 

@bgulanowski @davidmuzi

### CODE REVIEW

File: Mobile Buy SDK Sample Apps/Sample App Customers/Sample App Customers/AppDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    let channedID:  String = ""
+    
+    private(set) var client: BUYClient!
+
+    // ----------------------------------
+    //  MARK: - Application Launch -
+    //
+    func application(application: UIApplication, didFinishLaunchingWithOptions launchOptions: [NSObject: AnyObject]?) -> Bool {
+    
+        self.client = BUYClient(shopDomain: self.shopDomain, apiKey: self.apiKey, appId: self.appID)
```
Comment (davidmuzi) [inline Mobile Buy SDK Sample Apps/Sample App Customers/Sample App Customers/AppDelegate.swift]: appID not defined

Comment (dbart01) [inline Mobile Buy SDK Sample Apps/Sample App Customers/Sample App Customers/AppDelegate.swift]: Nice catch.

File: Mobile Buy SDK Sample Apps/Sample App Customers/Sample App Customers/AccountViewController.swift [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    
+    private var loginViewController:  LoginViewController!
+    private var signupViewController: SignupViewController!
+    
+    // ----------------------------------
+    //  MARK: - Segue -
+    //
+    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
+        switch segue.identifier {
+        case let login where login == "loginSegue":
```
Comment (bgulanowski) [inline Mobile Buy SDK Sample Apps/Sample App Customers/Sample App Customers/AccountViewController.swift]: These `login` and `signup` local values don't seem necessary. Can't you just say `case "loginSegue":`?

File: Mobile Buy SDK Sample Apps/Sample App Customers/Sample App Customers/BUYClient+Extensions.swift [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+extension BUYClient {
+    
+    static var sharedClient: BUYClient {
+        if let delegate = UIApplication.sharedApplication().delegate as? AppDelegate {
+            return delegate.client
+        }
+        fatalError("Could not retrieve shared BUYClient")
+    }
+}
```
Comment (bgulanowski) [inline Mobile Buy SDK Sample Apps/Sample App Customers/Sample App Customers/BUYClient+Extensions.swift]: Nitpick: missing newline. Xcode can issue a warning for this.

**Comment by bgulanowski** [comment]
Take a minute and go through the project settings to ensure all useful warnings are enabled. Or at the least, make sure the same warnings are enabled as on the other sample projects. As long as they make sense.

I would say to enable "treat all warnings as errors", but deprecations.

There are three warnings when we build, and we can probably prevent two of them, and possibly we should silence the third.

File: Mobile Buy SDK Sample Apps/Sample App Customers/Sample App Customers/AccountViewController.swift (hunk 2 above)
Comment (dbart01) [inline Mobile Buy SDK Sample Apps/Sample App Customers/Sample App Customers/AccountViewController.swift]: The only reason for that was to unwrap optional values but you're right, there's a simpler way.

**Comment by bgulanowski** [comment]
🚢  it!

**Pull request merged**